### DISCUSSION TITLE
How to use Immich to delete original files on device?

### DISCUSSION THREAD

**Discussion by 825i** [immich-app/immich#11047] [Q&A]
I have gone through and deleted hundreds of photos using Immich in my external library, just to find out that they are only deleted from Immich and not at all from my actual device. The whole reason I got this app was to be able to blaze through my photo timeline and mass delete images that I don't care about. I obviously want those images removed forever.

When checking the settings I can't see anywhere to actually delete the original files.

Is this seriously not a feature at all? What is the point of deleting photos in Immich if the junk files, duplicates, blurry photos etc. are all still taking up storage on my server/device?

**Comment by bo0tzz** [comment]
There's a few things at play here:
1. Files in external libraries are deleted from disk only if the library is not mounted readonly
2. Files in external libraries have no link to files on your phone - Immich only knows about that connection if they got uploaded through the app
3. If you delete a file from the mobile app, you get the option to also delete it locally on the phone
4. Deletes from the web/server are currently not propagated to the phone automatically. At the moment this is a concious choice, because we're dealing with important data and we don't want to risk a bug being able to accidentally delete all of it. Once Immich is more stable, we will implement a cleanup feature, see also #165.

**Reply by MCDELTAT** [reply]
> 3. If you delete a file from the mobile app, you get the option to also delete it locally on the phone

@bo0tzz Happy Holidays. Can you clarify "...you get the option..."? I just configured Immich today on TrueNAS. I'm running server v1.119.1 and the lastest app on iOS 18.2.0. I synced some ~1,000 photos to Immich. I then try to delete a few in the app. I see them go away on the server, they do not go away from my local device. Is that supposed to be a thing? Is it a setting somewhere?

**Reply by bo0tzz** [reply]
When you select assets you get a few different deletion options.

![image](https://github.com/user-attachments/assets/e5aa2f71-a950-42d3-b12b-9bde0ef675d5)

**Reply by MCDELTAT** [reply]
Gotcha. I’ll file this as a big against the iOS app then. 

On the iOS app those options aren’t available. The delete goes to the trash. When you select an individual photo in the trash to delete it, you get the warning “These items will be deleted permanently deleted from Immich and from your device.”

Going back to the iOS photos app, I can see all deleted images still there.

**Reply by ChocolateLoverRaj** [reply]
https://github.com/immich-app/immich/discussions/11047#discussioncomment-11705445 - Nice, I just learned about this. So you have to do the select multiple (even if it's just selecting 1) option to get the specific delete options. It would be nice if the delete button in the single photo view also showed you 3 options for deleting instead of just deleting from cloud and local.

**Reply by MCDELTAT** [reply]
@ChocolateLoverRaj I can confirm that I found those settings in the UI if you select things. However, it also isn't deleting from my phone in the server/app combo I'm using.

**Reply by serious-angel** [reply]
Just to clarify, since I haven't found such yet, is it possible to search and select all the already uploaded files on the phone app?

**Reply by JehandadK** [reply]
@serious-angel I think you have to select multiple files for this option to show up.

Also for dumb people like me:
- Delete : Delete from device and cloud
- Move to Trash: Will eventually delete from Cloud
- Delete from Device: Will **only** delete from device.

**Reply by serious-angel** [reply]
> ...you have to select multiple files for this option to show up...

Thank you for the response.  
Indeed, and this is how I do now, too, but there seem to be no option to select all 10,000 photos to process on the app, but per group per day only organized by Exif/Meta data, and hence the question, since currently, to select all, we may need to select each day separately.

**Reply by drybx** [reply]
> There's a few things at play here:
> 
> 1. Files in external libraries are deleted from disk only if the library is not mounted readonly

For some reason, that's not what's happening on my end. I’m running Immich in an LXC container (bare metal, no Docker) and  Immich definitely has write access to the folder where my external library is stored.  

I already emptied the Immich trash, but the deleted files still remain on my HDD instead of being removed.

Now, when I rescan my library, of course deleted photos reappear

**Reply by Jebble** [reply]
> 3. If you delete a file from the mobile app, you get the option to also delete it locally on the phone
> 4. Deletes from the web/server are currently not propagated to the phone automatically. At the moment this is a concious choice, because we're dealing with important data and we don't want to risk a bug being able to accidentally delete all of it. Once Immich is more stable, we will implement a cleanup feature, see also [[Feature] Mass remove already backed-up photos #165](https://github.com/immich-app/immich/issues/165).

Appreciate this is a late reply, but I've never seen this question asked. I've only started using Immich after the 2.0 release. I did just try this again and i'm not able to remove images from Immich and also have them deleted from my phone.

**Comment by 825i** [comment]
Ok I figured some of this out. I had mounted in Unraid the external library as Read Only.
After remounting as Read/Write. Deleting from the trash now does delete the original file.

However, I am having a second issue that despite the "/thumbs" DIR already having been mounted as Read/Write from the beginning. I am getting an EACCESS error that it cannot delete the thumb files when the image is deleted from the trash and the disk.

Is the process runner responsible for deleting the thumb file, running under a different user/process than what deletes the actual photo file?

**Comment by svenstaro** [comment]
I believe this should now be fixed as of immich v1.132.3 and https://github.com/immich-app/immich/pull/16732. Maybe change this to "answered"?

**Reply by Cheekie25** [reply]
It's still not the case for iOS, is it ?

**Comment by CiNcH83** [comment]
I also just deleted 2 photos from the Immich web interface. Those photos live on an external library which is being mounted `rw`:
```
    volumes:
      - ${IMMICH_HOME}/data:/data
      - ${DANIEL_LIB}:/Assets/Daniel
      - ${SABINE_LIB}:/Assets/Sabine
```

At least the external libs are not explicitly mounted `ro`. I also performed a "Scan All Libraries" afterwards. The files however haven't been deleted from the disk.

Is that intended behavior? Or will it only be deleted after 30 days?

Using `Immich 2.1.0`.

**Reply by CiNcH83** [reply]
I probably have to add that I am running as non-root:
```
services:
  immich-server:
    user: ${IMMICH_UID}:${IMMICH_GID}
    group_add:
      - ${DANIEL_GID}
      - ${SABINE_GID}
```

The photo files belong to my user which has been added with `group_add`:
```
Daniel@raspberrypi5:~/assets/2024/10 $ ls -lah
total 14M
drwxrwx---  2 Daniel Daniel 4.0K Oct 16 07:14 .
drwxrwx--- 13 Daniel Daniel 4.0K Nov  6  2024 ..
-rw-rw----  1 Daniel Daniel 1.9M Oct  2  2024 2024-10-02_17-49-05_IMG_0122.heic
-rw-rw----  1 Daniel Daniel 356K Oct  6  2024 2024-10-06_14-27-15_IMG_0145.jpg
-rw-rw----  1 Daniel Daniel 2.6M Oct 12  2024 2024-10-12_17-50-34_IMG_0151.heic
-rw-rw----  1 Daniel Daniel 2.6M Oct 12  2024 2024-10-12_17-50-37_IMG_0152.heic
-rw-rw----  1 Daniel Daniel 2.8M Oct 19  2024 2024-10-19_12-16-49_IMG_0175.heic
-rw-rw----  1 Daniel Daniel 3.6M Oct 26  2024 2024-10-26_15-05-31_IMG_0190.heic
```

I think that the file permissions are properly being set up.

**Reply by CiNcH83** [reply]
Never mind... got it! After emptying the trash, the files got deleted too.
